From b7c7f34e5cfa2d5bb49c4982531dc0641b15c021 Mon Sep 17 00:00:00 2001
From: Stas Boukarev <stassats@gmail.com>
Date: Mon, 1 Jun 2020 15:52:00 +0300
Subject: [PATCH] win32-os.c: remove reimplementations of printf, fprintf.

Doesn't fit well -fno-common and isn't a nice thing to do overall.

Reported by Jinsong Zhao.
---
 src/runtime/win32-os.c | 48 ------------------------------------------
 1 file changed, 48 deletions(-)

diff --git a/src/runtime/win32-os.c b/src/runtime/win32-os.c
index 2354bd40de..a92934dc52 100644
--- a/src/runtime/win32-os.c
+++ b/src/runtime/win32-os.c
@@ -679,54 +679,6 @@ int os_preinit(char *argv[], char *envp[])
 }
 #endif  /* LISP_FEATURE_SB_THREAD */
 
-
-#ifdef LISP_FEATURE_X86_64
-/* Windows has 32-bit 'longs', so printf...%lX (and other %l patterns) doesn't
- * work well with address-sized values, like it's done all over the place in
- * SBCL. And msvcrt uses I64, not LL, for printing long longs.
- *
- * I've already had enough search/replace with longs/words/intptr_t for today,
- * so I prefer to solve this problem with a format string translator. */
-
-/* There is (will be) defines for printf and friends. */
-
-static int translating_vfprintf(FILE*stream, const char *fmt, va_list args)
-{
-    char translated[1024];
-    unsigned i=0, delta = 0;
-
-    while (fmt[i-delta] && i<sizeof(translated)-1) {
-        if((fmt[i-delta]=='%')&&
-           (fmt[i-delta+1]=='l')) {
-            translated[i++]='%';
-            translated[i++]='I';
-            translated[i++]='6';
-            translated[i++]='4';
-            delta += 2;
-        } else {
-            translated[i]=fmt[i-delta];
-            ++i;
-        }
-    }
-    translated[i++]=0;
-    return vfprintf(stream,translated,args);
-}
-
-int printf(const char*fmt,...)
-{
-    va_list args;
-    va_start(args,fmt);
-    return translating_vfprintf(stdout,fmt,args);
-}
-int fprintf(FILE*stream,const char*fmt,...)
-{
-    va_list args;
-    va_start(args,fmt);
-    return translating_vfprintf(stream,fmt,args);
-}
-
-#endif
-
 int os_number_of_processors = 1;
 
 BOOL (*ptr_CancelIoEx)(HANDLE /*handle*/, LPOVERLAPPED /*overlapped*/);
